scrapy/extensions/logstats.py

```diff
@@ -35,18 +38,25 @@ def spider_opened(self, spider):
         self.task.start(self.interval)
 
     def log(self, spider):
-        items = self.stats.get_value('item_scraped_count', 0)
-        pages = self.stats.get_value('response_received_count', 0)
-        irate = (items - self.itemsprev) * self.multiplier
-        prate = (pages - self.pagesprev) * self.multiplier
-        self.pagesprev, self.itemsprev = pages, items
+        self.calculate_stats()
 
         msg = ("Crawled %(pages)d pages (at %(pagerate)d pages/min), "
                "scraped %(items)d items (at %(itemrate)d items/min)")
-        log_args = {'pages': pages, 'pagerate': prate,
-                    'items': items, 'itemrate': irate}
+        log_args = {'pages': self.pages, 'pagerate': self.prate,
+                    'items': self.items, 'itemrate': self.irate}
         logger.info(msg, log_args, extra={'spider': spider})
 
     def spider_closed(self, spider, reason):
         if self.task and self.task.running:
             self.task.stop()
+
+        self.calculate_stats()
+        self.stats.set_value('IPM', self.irate)
+        self.stats.set_value('RPM', self.prate)
```

Comment:
:lipstick:  I think `items_per_minute` and `requests_per_minute` would be a better choice here.

Reply:
Thanks @Gallaecio! I'll update this one along with the tests. Please don't merge this one yet as I'd like to have tests on this module. Marking it as `WIP:` at the moment.